dss/events/handlers/index.py

```diff
@@ -126,25 +130,110 @@ def add_index_data_to_elasticsearch(bundle_key, index_data, logger) -> None:
+        if not response:
             logger.debug(f"Creating new Elasticsearch index: {DSS_ELASTICSEARCH_INDEX_NAME}")
-            response = es_client.indices.create(DSS_ELASTICSEARCH_INDEX_NAME, body=None)
+            response = es_client.indices.create(DSS_ELASTICSEARCH_INDEX_NAME, body=index_mapping)
             logger.debug(f"Index creation response: {json.dumps(response, indent=4)}")
         else:
             logger.debug(f"Using existing Elasticsearch index: {DSS_ELASTICSEARCH_INDEX_NAME}", )
     except Exception as ex:
         logger.critical(f"Unable to create index {DSS_ELASTICSEARCH_INDEX_NAME}  Exception: {ex}")
+        raise
 
 
 def add_data_to_elasticsearch(bundle_id, index_data, logger) -> None:
     try:
         ElasticsearchClient.get(logger).index(index=DSS_ELASTICSEARCH_INDEX_NAME,
                                               doc_type=DSS_ELASTICSEARCH_DOC_TYPE,
                                               id=bundle_id,
-                                              body=json.dumps(index_data, indent=4))
-
+                                              body=json.dumps(index_data),
+                                              refresh=True)
     except Exception as ex:
         logger.error(f"Document not indexed. Exception: {ex}  Index data: {json.dumps(index_data, indent=4)}")
+        raise
+
+
+def find_matching_subscriptions(index_data, logger):
+    percolate_document = {
+        "query": {
+            "percolate": {
+                "field": "query",
+                "document_type": DSS_ELASTICSEARCH_DOC_TYPE,
+                "document": index_data
+            }
+        }
+    }
+    response = ElasticsearchClient.get(logger).search(
+        index=DSS_ELASTICSEARCH_INDEX_NAME,
+        body=percolate_document)
+    logger.debug("Found matching subscription count: %i", len(response['hits']['hits']))
+    subscription_ids = set()
+    for hit in response['hits']['hits']:
+        subscription_ids.add(hit["_id"])
+    return subscription_ids
+
+
+def process_notifications(bundle_id, subscription_ids, logger):
+    for subscription_id in subscription_ids:
+        try:
+            # TODO Batch this request
+            subscription = get_subscription(subscription_id, logger)
+            notify(subscription, bundle_id, logger)
+        except Exception as ex:
+            logger.error(f"Error occurred while processing subscription {subscription_id} for bundle {bundle_id}. {ex}")
+
+
+def get_subscription(subscription_id, logger):
+    subscription_query = {
+        "query": {
+            "ids": {
+                "type": DSS_ELASTICSEARCH_SUBSCRIPTION_TYPE,
+                "values": [subscription_id]
+            }
+        }
+    }
+    response = ElasticsearchClient.get(logger).search(
+        index=DSS_ELASTICSEARCH_SUBSCRIPTION_INDEX_NAME,
+        body=subscription_query)
+    if len(response['hits']['hits']) == 1:
+        return response['hits']['hits'][0]['_source']
```

Comment:
Yes, that is a nice pattern and will implement it.
I am going to open an new PR for this tomorrow as I need to leave for an interview and this PR is otherwise ready to merge.